unit-tests/run-unit-tests.hs

```diff
@@ -221,6 +227,47 @@ test_normalize_type = TestCase "normalize_type" [mod_file] $ do
                              ,"type instance Foo x = ()"]
           mod_file = "TEST_NormalizeType.hs"
 
+-- earlier versions of hint were accidentally overwriting the signal handlers
+-- for ^C and others.
+--
+-- note that hint was _not_ overwriting the signal handlers when the hint interpreter
+-- was itself executed inside the ghci interpreter. for this reason, this test always
+-- succeeds when executed from ghci and ghcid, regardless of whether the problematic
+-- behaviour has been fixed or not.
```

Comment:
Perhaps make the test fail (or at least show a clear "skipped test because ...") under ghci? Assuming that is possible.

Reply:
I don't know how to determine whether we're running under ghci or not. `GHC.runGhcT` must somehow detect it in order to not install the signal handlers, but how this is accomplished is not obvious from the code.

Reply:
Not a big deal - was just wondering if this was easy.